readthedocs/core/views/__init__.py

```diff
@@ -124,11 +124,33 @@ def server_error_404(request, exception=None, template_name='404.html'):  # pyli
             )
         else:
             return response
-    r = render(request, template_name)
+    r = render(request, template_name, context=_get_404_context(request))
     r.status_code = 404
     return r
 
 
+def _get_404_context(request):
+    from readthedocs.redirects.utils import project_and_path_from_request, language_and_version_from_path
+    project = None
+    language = None
+    version_slug = None
+
+    full_path = request.get_full_path()
```

Comment:
Does this suffer from the Django security issue they just patched? https://github.com/django/django/commit/1cd00fcf52d089ef0fe03beabd05d59df8ea052a

Reply:
We are already with the patched version https://github.com/rtfd/readthedocs.org/blob/ee2d7847240d05c4b7e494ec68b3fc2a654e8631/requirements/pip.txt#L18

Reply:
Right, but we aren't calling `urlquote`, which is the fix for the path issue.